### PR TITLE
fix (Change log group references to ILogGroup, don't create a second log group in aws-*-step-function)

### PR DESCRIPTION
**Pull request by biffgaut**
*Issue #, if available:*193

*Description of changes:*
This revealed a larger bug in the Step Functions constructs. The construct was not looking at stateMachineProps.logs to see if an existing log group had been provided and created a new log group every time. This means that there was no way to provide an existing log group. We decided to fix this even though it it may be a breaking change for outlying use cases.

The issue is that stateMachineProps.logs is ILogGroup, not LogGroup. Since ILogGroup cannot be upgraded to LogGroup, including stateMachineProps.logs as a source of log group meant that all references to Log Groups have to be converted to ILogGroup, including the properties of the constructs.

If you access the Log Group of a Step Functions construct, you may need to change the type to ILogGroup if you ever explicitly specify the type.

If you access the Log Group of a Step Functions construct and make any changes to the Log Group after the construct has been instantiated, your code will no longer work as ILogGroups cannot be altered. Your workaround will be to create a log group to your full specification before launching the Step Functions construct, the set the stateMachineProps.logs.destination to this new log group (a LogGroup object can be "downgraded" to an ILogGroup).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: a976fe08f25715b998e12ba38daa826963935c4d
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c6ee86f7-665b-46ff-bb82-d2e6f92e32f0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 39ba8efbfd5a3b5477011f50b24601db5731b3ba
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a3e1185a-7e98-49f0-ba8c-ed702299f43c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: eca4c16ac8c2b9dd643c6f07eafe383bfceb7328
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0e0e59b7-9d4d-49a9-8e09-2f36dfaf231e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-lambda-step-function/README.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ _Parameters_
 |:-------------|:----------------|-----------------|
 |lambdaFunction|[`lambda.Function`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda.Function.html)|Returns an instance of the Lambda function created by the pattern.|
 |stateMachine|[`sfn.StateMachine`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.StateMachine.html)|Returns an instance of StateMachine created by the construct.|
-|stateMachineLogGroup|[`logs.LogGroup`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-logs.LogGroup.html)|Returns an instance of the LogGroup created by the construct for StateMachine|
+|stateMachineLogGroup|[`logs.ILogGroup`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-logs.ILogGroup.html)|Returns an instance of the LogGroup created by the construct for StateMachine|
```
Comment (hayesry) [inline source/patterns/@aws-solutions-constructs/aws-lambda-step-function/README.md:77]: We should update the property description to reference ILogGroup as well (i.e. "Returns an instance of the **ILogGroup** created by the construct for the StateMachine.").

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 0b66e6d32fd1b497d19f3df116520e9fc8a06e5c
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=90969269-d2b1-4f37-8e14-3b3926e32186%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by hayesry — COMMENTED** [review]
Reviewed all files in the pull request. Helper function code looks good. Test coverage appears to be under 80% for the helper however. Recommend updating/organizing the test suite for the helper to more explicitly assert the outcomes for each of the four possibilities here (logGroup provided in stateMachineProps, logGroupProps not provided, logGroupProps provided w/o logGroupName, and logGroupProps provided w/ logGroupName). Overall nice, concise fix!

**Review by hayesry — APPROVED** [review]

**Comment by biffgaut** [comment]
Approved by Ryan, Hitendra is OOTO.

**Pull request merged**